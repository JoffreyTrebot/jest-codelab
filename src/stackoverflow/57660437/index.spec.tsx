import React from 'react';
import { Header } from './';
import { shallow, mount } from 'enzyme';
import * as requestService from './requestService';

jest.mock('./requestService.ts', () => {
  return {
    runningSince: jest.fn()
  };
});

describe('Header', () => {
  it('t1', done => {
    const mockedData = {
      data: {
        runningSince: new Date('2019-08-26T11:55:03.696Z').toISOString()
      }
    };
    (requestService.runningSince as jest.MockedFunction<typeof requestService.runningSince>).mockResolvedValueOnce(
      mockedData
    );
    const wrapper = shallow(<Header></Header>);
    setImmediate(() => {
      expect(wrapper.state()).toEqual({ startDate: new Date(mockedData.data.runningSince).toString() });
      expect(wrapper.text()).toEqual(new Date(mockedData.data.runningSince).toString());
      expect(wrapper).toMatchSnapshot();
      done();
    });
  });
});
