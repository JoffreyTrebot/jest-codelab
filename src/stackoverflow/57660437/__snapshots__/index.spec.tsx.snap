// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header t1 1`] = `
<div>
  Mon Aug 26 2019 19:55:03 GMT+0800 (China Standard Time)
</div>
`;
